docs/infrastructure/microk8s-quickstart.md

```diff
@@ -0,0 +1,136 @@
+
+`helm` is the client facing side, while `tiller` is the server/cluster facing side.
+
+## 2. Prepare your environment
+
+Prepare your environment to set up the microk8s cluster.
+For your infrastructure, you will need the following:
+
+1. A machine running Ubuntu 18.04 LTS.
+2. A postgres database and open firewall rule. ([Documentation can be found here.](../requirements.md))
+
+To start installing Codacy, you will need to:
+
+1. Establish a session (local or remote) onto the machine described in 1. above.
+
+Assuming that you are starting from a blank slate, the first step is to install microk8s. Otherwise, consider that Codacy is supported up to kubernetes 1.15. Please jump on to the section [Configuring microk8s](###4.-Configuring-microk8s) if you are on a compatible kubernetes version.
+
+## 3. Installing microk8s
+
+1. Make sure the machine is up to date and also has the `nfs-common` package installed
+
+   ```bash
+   sudo apt update && apt install nfs-common -y
+
+2. Install microk8s. Codacy is supported up to kubernetes 1.15. Therefore we recommend following the [official documentation on how to install microk8s](https://microk8s.io/docs/) with the `1.15/stable` channel.
+
+3. Allow privileged containers for microk8s
+
+   ```bash
+   sudo echo "--allow-privileged=true" >> /var/snap/microk8s/current/args/kube-apiserver
+   ```
+
+## 4. Configuring microk8s
+
+1. First, we must enable the following plugins on microk8s:
+
+    ```bash
+   microk8s.enable dns &&
+   microk8s.enable storage &&
+   microk8s.enable ingress &&
+   microk8s.status --wait-ready
+    ```
+
+2. Now that these plugins have been enabled, we should restart microk8s to ensure a proper bootstrap of the cluster
+
+    ```bash
+   microk8s.stop && microk8s.start && microk8s.status --wait-ready
+    ```
+
+    Further information on this subject can be [found here](https://github.com/ubuntu/microk8s/issues/493#issuecomment-498167435).
+
+3. The plugins are now enabled and the cluster bootstrapped. However, we must still wait for some microk8s internals (dns, http, and ingress) plugins to be ready. Failing to do so can result in pods entering a `CrashLoopBackoff` state:
+
+   ```bash
+    microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=kube-dns
+
+    microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=hostpath-provisioner
+
+    microk8s.kubectl wait -n ingress --for=condition=Ready pod -l name=nginx-ingress-microk8s
+    ```
+
+After these commands return successfully, we have ensured that dns, http, and nginx ingress are enabled and working properly inside the cluster.
+
+## 5. Installing Codacy
+
+### 1. Helm and Tiller
+
+First we must install `helm` onto the cluster. With `helm`, we can easily install charts and manage the lifecycle of the installed artifacts. One example is the ability to rollback a failed install.
+
+1. __Codacy supports up to `helm` version 2.16.3.__
```

Comment:
Should we add the command needed to install with this specific version?

Reply:
Installing Helm as part of preparing the environment to install Codacy is already mentioned here:

https://github.com/codacy/chart/blob/master/docs/index.md#preparing-to-install-codacy

We need to integrate the quickstart for the microk8s better with the main installation process in that page, though.

Reply:
done